### DISCUSSION TITLE
Support for newer MySQL authentication plugins

### DISCUSSION THREAD

**Discussion by Repox** [directus/directus#4468] [Archive]
Directus is currently using [this package](https://www.npmjs.com/package/mysql) for the MySQL connection. This package currently doesn't support MySQL8.0's `caching_sha2_password` authentication plugin, and a [pull request addressing this](https://github.com/mysqljs/mysql/pull/1962) hasn't been touched in over two years. 

I realize that a solution is to change the improved authentication to themysql_native_password plugin, but that just feels dirty (imho). 

Is there any effort in moving forward to a package like [mysql2](https://www.npmjs.com/package/mysql2) that seems to support the newer authentication plugins (and seems more maintained as well)?

I realize that it might not be a drop-in replacement package, but since MySQL 8 and the default authentication plugin has been around for some years now, this feels like an under prioritized issue.

**Comment by rijkvanzanten** [comment]
Yeah I agree 💯, however..

> Directus is currently using this package for the MySQL connection.

This is not 100% true, we rely on [Knex](http://knexjs.org) for the database abstraction, which in turn only supports the older default native password..

I've been looking to get support for `@mysql/xdevapi` in Knex itself, but that's a bigger project than it seems: https://github.com/knex/knex/issues/3963

**Reply by bobbyphilip** [reply]
At the moment, it looks like knex has added support for mysql2   Is this something which could be brought into directus as well?

**Reply by rijkvanzanten** [reply]
Last time we looked into this (about a year ago at this point), IIRC there was something preventing us from using mysql2. That being said, it's definitely worth looking into again now 👍🏻 I _should_ be a drop-in replacement, and knex has had support for a little while as well, so one would expect it should be doable?
